src/engraving/tests/playback/playbackeventsrendering_tests.cpp

```diff
@@ -2919,3 +2919,98 @@ TEST_F(Engraving_PlaybackEventsRendererTests, Trill_TiedNotes)
+    m_renderer.renderCountIn(score, firstMeasure->tick().ticks(), 0, m_defaultProfile, result, totalCountInDuration);
+
+    // [THEN] 4 quarter note events
+    EXPECT_EQ(result.size(), 4);
+    EXPECT_EQ(totalCountInDuration, QUARTER_NOTE_DURATION * 4);
+
+    timestamp_t expectedTimestamp = 0;
+
+    for (const auto& pair : result) {
+        EXPECT_EQ(pair.second.size(), 1);
+
+        for (const PlaybackEvent& event : pair.second) {
+            ASSERT_TRUE(std::holds_alternative<mpe::NoteEvent>(event));
+
+            const mpe::NoteEvent& noteEvent = std::get<mpe::NoteEvent>(event);
+            EXPECT_EQ(noteEvent.arrangementCtx().actualDuration, QUARTER_NOTE_DURATION);
+            EXPECT_EQ(noteEvent.arrangementCtx().actualTimestamp, expectedTimestamp);
+
+            expectedTimestamp += QUARTER_NOTE_DURATION;
+        }
+    }
+
+    // [WHEN] Render the same measure (4/4), but starting from the 2nd beat
+    result.clear();
+    totalCountInDuration = 0;
+    int secondBeatTick = firstMeasure->tick().ticks() + 960;
+    m_renderer.renderCountIn(score, secondBeatTick, 0, m_defaultProfile, result, totalCountInDuration);
+
+    // [THEN] 6 quarter note events
+    EXPECT_EQ(result.size(), 6);
+    EXPECT_EQ(totalCountInDuration, QUARTER_NOTE_DURATION * 6);
+
+    expectedTimestamp = 0;
+
+    for (const auto& pair : result) {
+        EXPECT_EQ(pair.second.size(), 1);
+
+        for (const PlaybackEvent& event : pair.second) {
+            ASSERT_TRUE(std::holds_alternative<mpe::NoteEvent>(event));
+
+            const mpe::NoteEvent& noteEvent = std::get<mpe::NoteEvent>(event);
+            EXPECT_EQ(noteEvent.arrangementCtx().actualDuration, QUARTER_NOTE_DURATION);
+            EXPECT_EQ(noteEvent.arrangementCtx().actualTimestamp, expectedTimestamp);
+
+            expectedTimestamp += QUARTER_NOTE_DURATION;
+        }
+    }
+
+    // [GIVEN] The next measure we want to render (3/8)
+    const Measure* secondMeasure = firstMeasure->nextMeasure();
+    ASSERT_TRUE(secondMeasure);
+
+    // [WHEN] Render the 2nd measure (3/8)
+    result.clear();
+    totalCountInDuration = 0;
+    m_renderer.renderCountIn(score, secondMeasure->tick().ticks(), 0, m_defaultProfile, result, totalCountInDuration);
+
+    // [THEN] 3 quaver note events
+    EXPECT_EQ(result.size(), 3);
+    EXPECT_EQ(totalCountInDuration, QUAVER_NOTE_DURATION * 3);
+
+    expectedTimestamp = 0;
+
+    for (const auto& pair : result) {
+        EXPECT_EQ(pair.second.size(), 1);
+
+        for (const PlaybackEvent& event : pair.second) {
+            ASSERT_TRUE(std::holds_alternative<mpe::NoteEvent>(event));
+
+            const mpe::NoteEvent& noteEvent = std::get<mpe::NoteEvent>(event);
+            EXPECT_EQ(noteEvent.arrangementCtx().actualDuration, QUAVER_NOTE_DURATION);
+            EXPECT_EQ(noteEvent.arrangementCtx().actualTimestamp, expectedTimestamp);
+
+            expectedTimestamp += QUAVER_NOTE_DURATION;
+        }
+    }
+
+    delete score;
+}
```

Comment:
Should we add a test for the situation that the user starts playback not at any beat, but just randomly at some point in time? (for example, they typed a minutes/seconds/milliseconds combination that doesn't coincide with a beat, or they just stopped playback at a random moment and now resume from that moment)